### PR TITLE
Fix pill clipping on first dictation after a monitor switch

### LINKED ISSUE #161: Bug: transcription pill is clipped at bottom after switching monitors
## Summary

On a dual-monitor setup, the floating transcription pill can be slightly cut off at the bottom when moving from a 1440p monitor to a 1080p monitor. After triggering transcription again on the same 1080p monitor, the pill corrects itself and is no longer clipped.

This looks like a first-placement-after-monitor-change bug, not a permanent layout/CSS issue.

## Reproduction steps

1. Use a dual-monitor setup with one 1440p display and one 1080p display.
2. Trigger a transcription while the active/dictation target is on the 1440p monitor.
3. Move focus/dictation target to the 1080p monitor.
4. Trigger transcription again.
5. Observe the pill on the 1080p monitor.
6. Trigger transcription one more time on the same 1080p monitor.

## Actual behavior

The first transcription pill shown after switching from the 1440p display to the 1080p display is slightly clipped/cut off at the bottom.

After another transcription on the same 1080p monitor, the clipping goes away.

## Expected behavior

The pill should be fully visible on the first transcription after changing monitors, with the same bottom gap and size it has after the second transcription.

## Initial investigation / likely cause

Relevant files:

- `src-tauri/src/pipeline/pill.rs`
- `src-tauri/src/pipeline/pill_position.rs`
- `src-tauri/src/core/window_geometry.rs`
- `src-tauri/src/main.rs`

The pill is created at runtime as a borderless transparent Tauri window with an initial inner size of `380x44`. Placement is handled by `pill_position.rs`, which:

- picks the target monitor from `target.display_point` using `monitor_from_point()`;
- falls back to the primary monitor when needed;
- converts the fixed pill size and bottom gap from logical points to physical pixels using the selected monitor's `scale_factor`;
- computes `y` from the selected monitor work area bottom minus pill height minus bottom gap;
- on Windows, applies the result using native `SetWindowPos()`.

The symptom specifically happening only on the first transcription after crossing monitors suggests one of these is happening:

1. **DPI / coordinate-space mismatch after monitor change**  
   The pill may still be using the previous monitor/window DPI context when `SetWindowPos()` is called for the first move to the 1080p display. The second transcription works because the window is already on the new display and the scale/position data has settled.

2. **Mixed logical vs physical coordinates**  
   `placement_for_monitor()` assumes the monitor work area and the scaled pill dimensions are in the same coordinate space. Verify what Tauri 2 returns for `Monitor::work_area()` on Windows in a mixed-resolution/mixed-DPI setup. If the work area is logical in some cases while the pill height/gap have been converted to physical pixels, the bottom `y` can be slightly too low.

3. **Outer window vs inner content sizing**  
   `apply_pill_placement()` computes the desired size from pill content dimensions, but the Windows branch applies it with `SetWindowPos()` to the native HWND. If the actual client area differs from the outer window rect immediately after a DPI/display transition, the WebView content can end up a few pixels short and appear clipped.

4. **Placement cache / stale invalidation edge case**  
   `next_pill_placement()` skips recomputation when `pill_placement_stale` is false and a cached placement exists. The normal hotkey path does mark placement stale before starting recording, but it is worth checking whether every path/state transition that can show the pill after monitor movement does the same, especially processing/error/handsfree paths.

## Debugging suggestions

Add temporary dev logging around pill placement:

- captured `target.display_point`;
- selected monitor work area, position, name if available, and scale factor;
- current pill `outer_position()`, `outer_size()`, `inner_size()`, and `scale_factor()` before placement;
- computed `PillPlacement { x, y, width, height }`;
- actual window/client rect after `SetWindowPos()` on Windows.

This should make it obvious whether the first 1080p placement is using the old monitor scale, the wrong work area, or an outer/client size mismatch.

## Possible fixes

- Recompute and apply pill placement after the window has actually moved/shown on the new monitor, not only before `ShowWindow()`.
- On Windows, after `SetWindowPos()`, query the actual window/client rect and do a second correction if the client height is less than the expected pill height.
- Invalidate `pill_placement_stale` on pill move/resize/scale-factor/display changes, not only at recording start.
- Clamp the final `y` against the selected monitor work area using the actual outer/client height after resize.
- Consider a small next-tick correction after monitor changes so WebView2/Tauri has processed the DPI transition before the final placement.

## Acceptance criteria

- The pill is fully visible on the first transcription after switching from a 1440p monitor to a 1080p monitor.
- The pill is also correct when switching from 1080p back to 1440p.
- Works with mixed Windows display scaling such as 100%/125%/150%.
- Repeated transcriptions on the same monitor do not drift, snap, or change the bottom gap.
- Add either a manual test note or a regression test around monitor work-area/scale-factor placement math where practical.

### PR DESCRIPTION
**Pull request by MONKE2525E**
## Thinking Path

> - Verenu is a Windows and macOS AI dictation app - hotkey hold -> audio capture -> transcription -> LLM cleanup -> text injection
> - Subsystem: pipeline - the floating dictation pill window's cross-monitor placement (`pipeline/pill.rs`, `pill_position.rs`, `pill_animation.rs`)
> - The pill was visibly clipped at the bottom on the first dictation shown after the active window moved to a different monitor (issue #161), but corrected itself on the next dictation on that same monitor
> - The codebase already had a safeguard for exactly this class of bug - an animated cross-monitor move added to avoid WebView2 swap-chain stutter on an instant cross-DPI resize - but it was gated on whether the pill was *currently visible*, not on whether it had *ever* been placed, so it never engaged for the very next dictation after `hide_pill` reset that flag
> - This pull request replaces that gate with one that tracks "has the pill ever had a real placement," so any reveal after the first one animates into a changed monitor target instead of jumping
> - The benefit is the pill reliably displays fully on the first dictation after switching monitors, confirmed on real 1440p/1080p hardware, with regression tests and optional diagnostics for future mixed-DPI edge cases

## What Changed

- Added `PILL_PLACED_ONCE` (`pill.rs`), which - unlike `PILL_VISUALLY_ACTIVE` - never resets when the pill goes idle, and used it to gate the animated cross-monitor move instead
- Extracted the animate/no-animate decision into a pure, unit-tested `should_animate_cross_monitor_move()` in `pill_position.rs`
- Added verbose-gated (`is_verbose()`) diagnostic logging at each placement decision point - resolved monitor/placement, the animate decision, and the actual on-screen rect immediately after each `SetWindowPos` call - to help diagnose any future mixed-DPI edge case
- Added a manual verification note to `Agent-Skills/SmokeTest.md` for real mixed-DPI hardware checks
- Added 3 unit tests: no animation before the pill's first placement, animation once placed with differing geometry, no animation once placed when geometry already matches

## Verification

- `cargo test --manifest-path src-tauri/Cargo.toml` and `npm run test:rust` - 272 passed, 1 pre-existing ignored, 0 failed (10 in `pipeline::pill_position::tests`, including the 3 new ones)
- `npm run lint` - `svelte-check` (0 errors) + `cargo clippy -- -D warnings` clean
- `npm test` (fast OnePyFone profile, 16 suites) - all passed
- Manual: on a real Windows dual-monitor setup (1440p @ 100% + 1080p), dictate on one monitor, switch focus to the other, dictate again - pill now displays fully on that first dictation; confirmed by the issue reporter
- No UI changes (backend-only), so no before/after screenshots apply

## Risks

- Windows-only change, scoped to the existing `#[cfg(target_os = "windows")]` animated-move path already used for cross-monitor moves; macOS pill placement is untouched
- New diagnostic logging is gated behind the existing verbose-logging flag (off by default) and only logs window coordinates/scale factors - never dictated text, prompts, or other sensitive content
- Low risk: corrects a stale boolean gate rather than the placement math itself, which was already covered by existing passing tests

> Check [`docs/ROADMAP.md`](../docs/ROADMAP.md) before opening feature PRs - work that overlaps with planned core changes may need to be redirected. See [`CLAUDE.md`](../CLAUDE.md) for architecture and contribution guidance.

## Model Used

- Claude Sonnet 4.6 (Anthropic), via Claude Code - standard tool use, no extended thinking mode

## Checklist

- [x] Thinking path traces from Verenu context down to this specific change
- [x] Model used is specified (provider + version)
- [x] Checked `docs/ROADMAP.md` - this PR does not duplicate planned work
- [x] `npm run check` passes (TypeScript)
- [x] `npm run lint` passes (ESLint + Clippy)
- [x] `npm run test:rust` passes
- [x] Smoke tests pass (see `Agent-Skills/SmokeTest.md` for commands)
- [x] Tests added or updated where applicable
- [ ] UI changes include before/after screenshots
- [x] No API keys, secrets, or personal data in code or logs
- [ ] If version bumped: all three files updated together (`package.json`, `src-tauri/tauri.conf.json`, `src-tauri/Cargo.toml`)
- [x] Relevant documentation updated to reflect changes
- [x] Risks documented above

Fixes #161

### CODE REVIEW

File: src-tauri/src/pipeline/pill.rs [hunk 1]
```diff
@@ -25,10 +25,18 @@ static REVEAL_GEN: AtomicU64 = AtomicU64::new(0);
...
 static PILL_VISUALLY_ACTIVE: AtomicBool = AtomicBool::new(false);
 
+/// Whether the pill has ever had a real, monitor-resolved placement applied
+/// in this process. `false` only for the very first `show_pill_msg` call —
+/// after that, even a reveal that follows a `hide_pill` idle cycle still has
+/// real (if stale) geometry on screen, so a monitor change found on that
+/// reveal is still worth animating into rather than jumping. Unlike
+/// `PILL_VISUALLY_ACTIVE`, this never resets back to `false`.
+#[cfg(target_os = "windows")]
+static PILL_PLACED_ONCE: AtomicBool = AtomicBool::new(false);
```
[the remark this answers is not in the document]
Comment (MONKE2525E) [reply src-tauri/src/pipeline/pill.rs:38]: Good catch, this was a real bug. Fixed it, but differently than the suggestion: instead of making `PILL_PLACED_ONCE` `pub(crate)` and setting it from `apply_pill_placement`, I moved the `.swap(true, ...)` call out of the `if let Some(current) = current_placement(...)` block entirely so it runs unconditionally in `show_pill_msg` as soon as we have a resolved placement to apply - regardless of whether `current_placement()` itself succeeds. Same fix, but keeps the flag scoped to `pill.rs` instead of widening its visibility across modules.

File: src-tauri/src/pipeline/pill_position.rs [hunk 2]
```diff
@@ -172,6 +204,17 @@ pub(super) fn apply_pill_placement<R: Runtime>(
...
+                // Temporary diagnostic for issue #161: compare the placement
+                // we asked for against what Tauri reads back from the HWND
+                // immediately after SetWindowPos returns, to catch any
+                // post-call DPI-driven override of our explicit size/position.
+                if crate::system::logger::is_verbose() {
+                    let actual = current_placement(pill);
+                    log::debug!(
+                        "pill apply_pill_placement (sync): needs_resize={needs_resize} needs_reposition={needs_reposition} intended={placement:?} actual_after_setwindowpos={actual:?}"
+                    );
+                }
```
[the remark this answers is not in the document]
Comment (MONKE2525E) [reply src-tauri/src/pipeline/pill_position.rs:220]: Same bug, same fix - see my reply on the pill.rs:38 thread. Went with setting `PILL_PLACED_ONCE` true in `show_pill_msg` itself rather than here in `apply_pill_placement`, so this function (the generic placement-applier, also used by the non-cross-monitor path) doesn't need to know about a flag that's really about `show_pill_msg`'s own "is this the first reveal" decision.

**Comment by MONKE2525E** [comment]
@gemini-code-assist Thanks for the review — both inline comments were the same real bug, fixed in the latest commit:

**`PILL_PLACED_ONCE` staying `false` past the first reveal** — you were right. The flag was only set inside `if let Some(current) = current_placement(...)`, and `current_placement()` can return `None` on the very first call (window not yet settled). When that happened, the flag never flipped, so the *next* reveal would also read `already_placed = false` and skip animating — reproducing the original clipping bug, just shifted one call later. I moved the flag swap out of that `if let` so it's set unconditionally as soon as a placement has been resolved, regardless of whether `current_placement()` succeeds. I went with that instead of the suggested `pub(crate)` + setting it inside `apply_pill_placement`, since that would make a generic placement-applying function aware of a flag that's really about `show_pill_msg`'s own "is this the first reveal" decision — keeping it self-contained in one function seemed cleaner. Functionally equivalent fix either way.

**Unrelated but also fixed**: CI's macOS Rust check was failing — `should_animate_cross_monitor_move` had no caller on non-Windows builds (its only call site is Windows-gated), which `-D warnings` treats as a hard `dead_code` error there. Gated it the same way the existing `current_placement` already was.

Both threads replied to and resolved. Would appreciate a fresh pass over the current diff.

File: src-tauri/src/pipeline/pill_position.rs [hunk 3]
```diff
@@ -115,6 +126,30 @@ pub(super) fn position_changed(current: i32, desired: i32) -> bool {
...
+pub(super) fn should_animate_cross_monitor_move(
+    already_placed: bool,
+    current: PillPlacement,
+    target: PillPlacement,
+) -> bool {
+    already_placed
+        && (dimension_changed(current.width as f64, target.width as f64)
+            || dimension_changed(current.height as f64, target.height as f64)
+            || position_changed(current.x, target.x)
+            || position_changed(current.y, target.y))
```
[the remark this answers is not in the document]
Comment (MONKE2525E) [reply src-tauri/src/pipeline/pill_position.rs:150]: Good simplification, applied as suggested - width/height in PillPlacement are already i32, so position_changed handles them natively without the f64 round-trip. dimension_changed is still needed for apply_pill_placement's u32-based size comparisons, so kept that one as-is.

**Comment by MONKE2525E** [comment]
@gemini-code-assist Applied your width/height simplification suggestion (reusing `position_changed` instead of casting to `f64` for `dimension_changed`) and pushed it. All CI checks are green now too. Let me know if there's anything else.

**Comment by MONKE2525E** [comment]
@gemini-code-assist /gemini review
